Ouch delete img and iso extension if image is archive and you try to upack it

### Version

ouch 0.4.1

### Description

Openwrt and few other projects compress their images (iso or img) to something like `xxx.iso.zip` or `yyy.img.gz`.
If I try decompress them, `ouch` will delete all extensions, so it will be file named `xxx` and not `xxx.iso` or `yyy` and not `yyy.img`.

Is there workaround for it? I tried to prevent it with
```
ouch decompress yyy.img.gz yyy.img
```
but it throw error
```
[ERROR] failed to canonicalize path `yyy.img`
 - File not found
```

### Current Behavior

_No response_

### Expected Behavior

_No response_

### Additional Information

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,10 @@ Categories Used:
 
 - Multi-threaded compression for gzip and snappy using gzp [\#348](https://github.com/ouch-org/ouch/pull/348) ([figsoda](https://github.com/figsoda))
 
+### Bug fixes
+
+- Fix handling of unknown extensions during decompression [\#355](https://github.com/ouch-org/ouch/pull/355) ([figsoda](https://github.com/figsoda))
+
 ## [0.4.1](https://github.com/ouch-org/ouch/compare/0.4.0...0.4.1)
 
 ### New Features

--- a/src/extension.rs
+++ b/src/extension.rs
@@ -113,11 +113,12 @@ fn to_extension(ext: &[u8]) -> Option<Extension> {
     ))
 }
 
-fn split_extension<'a>(name: &mut &'a [u8]) -> Option<&'a [u8]> {
+fn split_extension(name: &mut &[u8]) -> Option<Extension> {
     let (new_name, ext) = name.rsplit_once_str(b".")?;
     if matches!(new_name, b"" | b"." | b"..") {
         return None;
     }
+    let ext = to_extension(ext)?;
     *name = new_name;
     Some(ext)
 }
@@ -149,7 +150,7 @@ pub fn separate_known_extensions_from_name(path: &Path) -> (&Path, Vec<Extension
     };
 
     // While there is known extensions at the tail, grab them
-    while let Some(extension) = split_extension(&mut name).and_then(to_extension) {
+    while let Some(extension) = split_extension(&mut name) {
         extensions.insert(0, extension);
     }
 

